librz/analysis/var_global.c

```diff
@@ -0,0 +1,419 @@
+	if (!glob) {
+		eprintf("No such global variable!\n");
+		return false;
+	}
+	RZ_FREE(glob->name);
+	glob->name = strdup(newname);
+	return ht_pp_update_key(analysis->ht_global_var, old_name, newname);
+}
+
+/**
+ * \brief Set the comment of the global variable
+ * 
+ * \param analysis RzAnalysis
+ * \param name The name of the global variable to set
+ * \param comment The comment to set
+ * \return true if succeed
+ */
+RZ_API bool rz_analysis_var_global_set_comment(RzAnalysis *analysis, const char *name, RZ_NONNULL const char *comment) {
+	rz_return_val_if_fail(analysis && name && comment, false);
+	RzAnalysisVarGlobal *glob = rz_analysis_var_global_get_byname(analysis, name);
+	if (!glob) {
+		return false;
+	}
+	RZ_FREE(glob->comment);
+	glob->comment = strdup(comment);
+	return true;
+}
+
+/**
+ * \brief Set the type of the global variable
+ * 
+ * \param glob Global variable instance
+ * \param type The type to set. RzType*
+ * \return void
+ */
+RZ_API void rz_analysis_var_global_set_type(RzAnalysisVarGlobal *glob, RzType *type) {
+	rz_return_if_fail(glob && type);
+	glob->type = type;
+}
+
+static st64 var_access_cmp(ut64 x, char *y) {
+	return x - (ut64)((RzAnalysisVarGlobal *)y)->addr;
+}
+
+/**
+ * \brief Set the accesses of the global variable
+ * 
+ * \param analysis RzAnalysis
+ * \param glob Global variable instance
+ * \param reg Register
+ * \param access_addr Address of access
+ * \param access_type Type of access
+ * \param stackptr Stack pointer
+ * \return void
+ */
+RZ_API void rz_analysis_var_global_set_access(RzAnalysis *analysis, RzAnalysisVarGlobal *glob, const char *reg, ut64 access_addr, int access_type, st64 stackptr) {
+	rz_return_if_fail(glob);
+	st64 offset = (st64)access_addr - (st64)glob->addr;
+
+	// accesses are stored ordered by offset, use binary search to get the matching existing or the index to insert a new one
+	size_t index;
+	rz_vector_lower_bound(&glob->accesses, offset, index, var_access_cmp);
+	RzAnalysisVarAccess *acc = NULL;
+	if (index < glob->accesses.len) {
+		acc = rz_vector_index_ptr(&glob->accesses, index);
+	}
+	if (!acc || acc->offset != offset) {
+		acc = rz_vector_insert(&glob->accesses, index, NULL);
+		acc->offset = offset;
+		acc->type = 0;
+	}
+
+	acc->type |= (ut8)access_type;
+	acc->stackptr = stackptr;
+	acc->reg = rz_str_constpool_get(&analysis->constpool, reg);
+}
+
+/**
+ * \brief Remove the access at the address of global variable
+ * 
+ * \param glob Global variable instance
+ * \param address Where to delete
+ * \return void
+ */
+RZ_API void rz_analysis_var_global_remove_access_at(RzAnalysisVarGlobal *glob, ut64 address) {
+	rz_return_if_fail(glob);
+	st64 offset = (st64)address - (st64)glob->addr;
+	size_t index;
+	rz_vector_lower_bound(&glob->accesses, offset, index, var_access_cmp);
+	if (index >= glob->accesses.len) {
+		return;
+	}
+	RzAnalysisVarAccess *acc = rz_vector_index_ptr(&glob->accesses, index);
+	if (acc->offset == offset) {
```

Comment:
can `acc` be NULL ? if yes, apply the suggestion.
```suggestion
	if (acc && acc->offset == offset) {
```